cedar-example-use-cases/discord/README.md

```diff
@@ -0,0 +1,23 @@
+# Cedar Discord Example
+
+This repository contains a limited model of the [discord permissions system](https://support.discord.com/hc/en-us/articles/206029707-Setting-Up-Permissions-FAQ).
+
+The file `src/main.rs` sets up example users and demonstrates they have different permissions based on the different roles.
+Discord is interesting because users may have multiple roles and some users may also set the permissions of other roles dynamically.
+In this example, we implement this functionality by using
+Cedar's parent system to build a DAG that looks something like this:
+
+```
+ Permission::"SendMessage"    Permission::"KickMember" 
+         ▲        ▲                     ▲              
+         │        └───────────────────┐ │              
+         │                            │ │              
+   Role::"everyone"              Role::"admin"         
+         ▲                             ▲
+         │                             │
+   User::"yihong"                User::"oflatt"
```

Comment:
IIUC Discord correctly, a role is assigned permissions _for a particular channel_. So I don't see how it makes sense for the hierarchy to just relate `Role` to `Permission`.

I would think you need to set it up so that you have `Channel` resource, and that when the operator assigns permissions to a channel for a role, you basically create an ad hoc policy that expresses those permissions.

You also seem to be missing the concepts of `Category` for channels (which can be "synced" or not), and the fact that permissions can apply to _all_ channels (server wide). I would think you need a `Server` object which channels are `in`.

Reply:
Great ideas!
I suggest that we merge this PR without channels (I'll re-name Channel to Server for now)
I'll submit a follow-up PR that introduces channels, and another for categories.